klog.go

```diff
@@ -769,7 +769,7 @@ func (l *loggingT) printWithFileLine(s severity, logr logr.Logger, filter LogFil
 	if buf.Bytes()[buf.Len()-1] != '\n' {
 		buf.WriteByte('\n')
 	}
-	l.output(s, logr, buf, file, line, alsoToStderr)
+	l.output(s, logr, buf, 0, file, line, alsoToStderr)
```

Comment:
Passing '0' is rather conservative here. My own conclusion in https://github.com/kubernetes/klog/pull/234 was that we need to skip at least one level for the standard logging code which invokes this function.

But I admit that I didn't test that. Perhaps it would be worthwhile to add a test case so that we don't need to speculate?

Reply:
this specific occurrence is in the `printWithFileLine` method https://github.com/kubernetes/klog/blob/2f2fd5b17f08fc1f6df20b51ebe9f6f014c63d8f/klog.go#L754-L773

which is only called here: https://github.com/kubernetes/klog/blob/2f2fd5b17f08fc1f6df20b51ebe9f6f014c63d8f/klog.go#L1232-L1234

ill add a test that ensures the right depth is logged when using the standard go logging framework with a logr backend

Reply:
have something working, but the offset depends on how the go `log` package implements `Print` etc.

Reply:
Yes, I was expecting this to not work correctly in all cases. My point was more that 0 is likely to not be correct in any case and should be replaced with something higher that at least is right in some cases.